pkg/kubectl/genericclioptions/builder_flags.go

```diff
@@ -14,68 +14,95 @@ See the License for the specific language governing permissions and
+			Usage:     "identifying the resource.",
+			Filenames: &filenames,
+			Recursive: bool_ptr(true),
 		},
+
+		LabelSelector: str_ptr(""),
+		FieldSelector: str_ptr(""),
+		AllNamespaces: bool_ptr(false),
+
+		All:   bool_ptr(false),
+		Local: bool_ptr(false),
 	}
 }
 
 // AddFlags registers flags for finding resources
 func (o *ResourceBuilderFlags) AddFlags(flagset *pflag.FlagSet) {
-	flagset.StringSliceVarP(&o.FilenameOptions.Filenames, "filename", "f", o.FilenameOptions.Filenames, "Filename, directory, or URL to files identifying the resource.")
-	annotations := make([]string, 0, len(resource.FileExtensions))
-	for _, ext := range resource.FileExtensions {
-		annotations = append(annotations, strings.TrimLeft(ext, "."))
-	}
-	flagset.SetAnnotation("filename", cobra.BashCompFilenameExt, annotations)
-	flagset.BoolVar(&o.FilenameOptions.Recursive, "recursive", o.FilenameOptions.Recursive, "Process the directory used in -f, --filename recursively. Useful when you want to manage related manifests organized within the same directory.")
+	o.FileNameFlags.AddFlags(flagset)
 
-	flagset.StringVarP(&o.LabelSelector, "selector", "l", o.LabelSelector, "Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)")
-	flagset.StringVar(&o.FieldSelector, "field-selector", o.FieldSelector, "Selector (field query) to filter on, supports '=', '==', and '!='.(e.g. --field-selector key1=value1,key2=value2). The server only supports a limited number of field queries per type.")
-	flagset.BoolVar(&o.AllNamespaces, "all-namespaces", o.AllNamespaces, "If present, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.")
+	if o.LabelSelector != nil {
+		flagset.StringVarP(o.LabelSelector, "selector", "l", *o.LabelSelector, "Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)")
+	}
+	if o.FieldSelector != nil {
+		flagset.StringVar(o.FieldSelector, "field-selector", *o.FieldSelector, "Selector (field query) to filter on, supports '=', '==', and '!='.(e.g. --field-selector key1=value1,key2=value2). The server only supports a limited number of field queries per type.")
+	}
+	if o.AllNamespaces != nil {
+		flagset.BoolVar(o.AllNamespaces, "all-namespaces", *o.AllNamespaces, "If present, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.")
+	}
 }
 
 // ToBuilder gives you back a resource finder to visit resources that are located
-func (o *ResourceBuilderFlags) ToBuilder(restClientGetter genericclioptions.RESTClientGetter, resources []string) ResourceFinder {
+func (o *ResourceBuilderFlags) ToBuilder(restClientGetter RESTClientGetter, resources []string) ResourceFinder {
 	namespace, enforceNamespace, namespaceErr := restClientGetter.ToRawKubeConfigLoader().Namespace()
 
+	labelSelector := ""
+	if o.LabelSelector != nil {
```

Comment:
No, you need to use these ifs to decide whether to call the builder methods at  all.

```go
if o.labelselector != nil{
 builder.LabelSelectorParam()
}
```